packages/medusa/src/strategies/product-export.ts

```diff
@@ -0,0 +1,670 @@
+import { Request, Response } from "express"
+import { EntityManager } from "typeorm"
+import { AbstractBatchJobStrategy, IFileService } from "../interfaces"
+import { Product, ProductVariant } from "../models"
+import { BatchJobService, ProductService } from "../services"
+import { BatchJobStatus } from "../types/batch-job"
+import { defaultAdminProductRelations } from "../api/routes/admin/products"
+import { ProductRepository } from "../repositories/product"
+import { FindConfig } from "../types/common"
+import { MedusaError } from "medusa-core-utils/dist"
+import { AdminPostBatchesReq } from "../api/routes/admin/batch/create-batch-job"
+import { prepareListQuery } from "../utils/get-query-config"
+
+type Context = {
+  listConfig: FindConfig<Product>
+  filterableFields: Record<string, unknown>
+  offset: number
+}
+
+type InjectedDependencies = {
+  manager: EntityManager
+  batchJobService: BatchJobService
+  productService: ProductService
+  fileService: IFileService<any>
+  productRepository: typeof ProductRepository
+}
+
+type ColumnSchemaEntity = "product" | "variant"
+
+type ColumnSchemaDescriptor =
+  | {
+      accessor: (product: Product) => string
+      entityName: Extract<ColumnSchemaEntity, "product">
+    }
+  | {
+      accessor: (variant: ProductVariant) => string
+      entityName: Extract<ColumnSchemaEntity, "variant">
+    }
+
+export default class ProductExportStrategy extends AbstractBatchJobStrategy<ProductExportStrategy> {
+  public static identifier = "product-export-strategy"
+  public static batchType = "product-export"
+
+  public defaultMaxRetry = 3
+
+  protected manager_: EntityManager
+  protected transactionManager_: EntityManager | undefined
+
+  protected readonly batchJobService_: BatchJobService
+  protected readonly productService_: ProductService
+  protected readonly fileService_: IFileService<any>
+  protected readonly productRepository_: typeof ProductRepository
+
+  protected readonly relations_ = [...defaultAdminProductRelations]
```

Comment:
suggestion, very-nit(non-blocking): Rename to be more descriptive
```suggestion
  protected readonly defaultRelations_ = [...defaultAdminProductRelations]
```